### PR TITLE
Use eslint and prettier and no implicit any

### PR DESCRIPTION
**Pull request by linyows**

### CODE REVIEW

File: src/client.interface.ts [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
 
-export interface IResult {
+export interface Result {
   ErrCode?: string
   ErrInfo?: string
 }
+
+export type UnknownParams = {
+  [key: string]: qs.UnknownFacade
+}
```
Comment (dojineko) [inline src/client.interface.ts:19]: plz EOF blank line

File: src/client.ts [hunk 2]
```diff
@@ -1,36 +1,39 @@
...
-    const parsed: any = qs.parse(await res.text(), {decoder: decodeURIComponent});
+    const parsed = qs.parse(await res.text(), {decoder: decodeURIComponent})
 
     if (!res.ok || this.isError(parsed)) {
       throw new BadRequest(`Bad Request: ${pathname}`).
         setResponse(res).parseError(parsed)
     }
 
-    return parsed
+    return parsed as any as U
```
Comment (wataru-script) [inline src/client.ts:33]: [ここ](https://github.com/pepabo/gmopg/pull/30/files#diff-3cafb86a5b0a9d0916a7fed543a1fb53L23)でanyを許容してにUにcastでいいかも

File: src/client.ts [hunk 3]
```diff
@@ -1,36 +1,39 @@
...
-  public async post(pathname: string, data: any): Promise<any> {
+  public async post<T, U>(pathname: string, data: T): Promise<U> {
     const res: Response = await fetch(this.config.baseUrl + pathname, {
       method: 'POST',
       body: qs.stringify(data, {encode: false}),
       ...this.config.http,
     })
 
-    const parsed: any = qs.parse(await res.text(), {decoder: decodeURIComponent});
+    const parsed = qs.parse(await res.text(), {decoder: decodeURIComponent})
```
Comment (wataru-script) [inline src/client.ts:26]:
```suggestion
    const parsed: any = qs.parse(await res.text(), {decoder: decodeURIComponent})
```

File: src/client.ts (hunk 2 above)
Comment (wataru-script) [inline src/client.ts:33]:
```suggestion
    return parsed as U
```

**Pull request merged**